### PR TITLE
feat: varTheme based theming

### PR DESCRIPTION
**Pull request by jhoward-eightfold**
## SUMMARY:
Initial implementation of exposing css var based via config provider. This will be extended to other components and leveraged to customize visual styling without having to write css.

## GITHUB ISSUE (Open Source Contributors)
NA

## JIRA TASK (Eightfold Employees Only):
NA

## CHANGE TYPE:

-   [ ] Bugfix Pull Request
-   [x] Feature Pull Request

## TEST COVERAGE:

-   [ ] Tests for this change already exist
-   [ ] I have added unittests for this change

## TEST PLAN:
Manually testing in Storybook to ensure visuals are maintained in the button component suite. Additionally, testing configProvider updates in Storybook to ensure json based overriding of the css vars get applied as desired. Screenshot shows simple updates to remove pill shaped rounding and make primary buttons pink

Screen Shot 2022-12-02 at 2 00 22 PM

### CODE REVIEW

File: src/styles/themes/_default-theme.scss [hunk 1]
```diff
@@ -181,7 +181,7 @@
         Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif;
     --font-size: 16px;
 
-    /* ------ Tabs theme ------ */
+    // ------ Tabs theme ------
```
Comment (jhoward-eightfold) [inline src/styles/themes/_default-theme.scss:184]: `/* */` style comments remain in css output
`//` style comments get dropped in output

File: src/components/ConfigProvider/ConfigProvider.stories.tsx [hunk 2]
```diff
@@ -201,6 +202,32 @@ const ThemedComponents: FC = () => {
                     />
                 </div>
             </Stack>
+            <div>
+                <TextArea
+                    enableExpand
+                    labelProps={{ text: 'Var Theme (updates onBlur)' }}
```
Comment (dkilgore-eightfold) [inline src/components/ConfigProvider/ConfigProvider.stories.tsx:208]: An apply theme button would be more straight forward as opposed to manually adding textarea content onBlur? Another idea would be to a few forms in modals triggered by buttons - like [Custom button theme] [Custom table theme]

The Modal would contain a list inputs with the current css var values, then the person may pick and choose which to manually override at a glance as opposed to figuring out what the css vars are via the web inspector. The form would then send that back to the text area and apply the theme.

Something to consider.

File: src/components/ConfigProvider/Theming/Theming.types.ts [hunk 3]
```diff
@@ -35,6 +42,7 @@ export interface OcBaseTheme {
     errorColor?: Value;
     tabsTheme?: TabsTheme;
     navbarTheme?: NavbarTheme;
+    varTheme?: VarTheme;
```
Comment (dkilgore-eightfold) [inline src/components/ConfigProvider/Theming/Theming.types.ts:45]: Might also consider adding buttonTheme and tableTheme, etc. That way we have the potential of global varTheme and scoped component themes. Also ensures previous work scales.

File: src/styles/themes/_default-theme.scss [hunk 4]
```diff
@@ -238,5 +238,255 @@
...
+    // ------ Table theme ------
+
+    // ------ Generic button vars ------
+    --button-border-radius: 4px;
+    --button-border-radius-round-shape: 50%;
+    --button-pill-shape-border-radius: 24px;
+    // TODO: button text/font related vars
+    // ------ Generic button vars ------
+
+    // ------ Default Button theme ------
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:303]: These look familiar 😃

**Review by dkilgore-eightfold — APPROVED** [review]

File: src/components/ConfigProvider/ConfigProvider.stories.tsx (hunk 2 above)
Comment (jhoward-eightfold) [reply src/components/ConfigProvider/ConfigProvider.stories.tsx]: per office hours discussion, implemented the dialog to eliminate hacky onBlur implementation and have linked to the scss file for reference as to whats available

File: src/components/ConfigProvider/Theming/Theming.types.ts (hunk 3 above)
Comment (jhoward-eightfold) [reply src/components/ConfigProvider/Theming/Theming.types.ts:45]: per office hour discussion, i'm not sure I want to carry this existing pattern forward, implementation of it has more touchpoints/complications, including in Eightfold configurations.

**Pull request merged**